spec/example_app/db/migrate/20170331144723_create_countries.rb

```diff
@@ -0,0 +1,12 @@
+class CreateCountries < ActiveRecord::Migration
+  def change
+    create_table :countries do |t|
+      t.string :code
```

Comment:
```ruby
t.string :code, null: false, index: { unique: true }
```

That way you don't need the `add_index` at the bottom. Also, since you're validating at the application level, these should be `null: false` as well.